src/ir/comp.rs

```diff
@@ -18,6 +18,51 @@ impl Context {
     }
 }
 
+/// Queries corresponding signature [Event]s and [Port]s for their invoke counterparts
+impl Context {
+    /// Gets the [PortIdx] of the signature port of the invoked component corresponding
+    /// to the given [PortIdx] in the invocation.
+    pub fn get_component_port(
+        &self,
+        comp: &Component,
+        idx: PortIdx,
+    ) -> PortIdx {
```

Comment:
As I mentioned in our sync discussion, this `PortIdx` value is not safe to be used within a component; it is only meaningful for the component that defines the port.